var/spack/repos/builtin/packages/scorep/package.py

```diff
@@ -72,6 +74,7 @@ def install(self, spec, prefix):
                 "--with-papi-header=%s" % spec['papi'].prefix.include,
                 "--with-papi-lib=%s" % spec['papi'].prefix.lib,
                 "--enable-shared",
+                "--without-shmem" if '~shmem' in spec else '',
```

Comment:
@sknigh can you get this one?

@adamjstewart: is it worth making a free function or something for autotools packages to handle this? e.g.:

```python
def with_or_without(self, variant, arg=None):
    if arg is None:
        arg = variant.strip('~-+')
    return '--with{0}-{1}'.format('' if variant in spec else 'out', arg)
```

usage:

```python
    "--with-papi-lib=%s" % spec['papi'].prefix.lib,
    "--enable-shared",
    with_or_without('+shmem'),
    # etc.
```